medmodels/medrecord/_overview.py

```diff
@@ -1,109 +1,150 @@
-                ]
-            elif schema[attribute][1] == AttributeType.Categorical:
-                attribute_info = [
-                    _extract_string_attribute_info(
-                        attribute_series=attribute_values,
-                        long_string_suffix="categories",
-                        short_string_prefix="Categories",
-                    )
-                ]
+                attribute_info = _extract_temporal_attribute_info(attribute_values)
             else:
-                attribute_info = [
-                    _extract_string_attribute_info(attribute_series=attribute_values)
-                ]
-
-        ## Without Schema
+                attribute_info = _extract_string_attribute_info(
+                    attribute_series=attribute_values,
+                    long_string_suffix="categories",
+                    short_string_prefix="Categories",
+                )
+        # Without Schema
         else:
             if attribute_values.dtype.is_numeric():
-                attribute_info = [
-                    f"min: {attribute_values.min()}",
-                    f"max: {attribute_values.max()}",
-                    f"mean: {attribute_values.mean():.{decimal}f}",
-                ]
+                attribute_info = _extract_numeric_attribute_info(attribute_values)
             elif attribute_values.dtype.is_temporal():
-                attribute_info = [
-                    f"min: {min(attribute_values).strftime('%Y-%m-%d %H:%M:%S')}",
-                    f"max: {max(attribute_values).strftime('%Y-%m-%d %H:%M:%S')}",
-                ]
+                attribute_info = _extract_temporal_attribute_info(attribute_values)
             else:
-                attribute_info = [
-                    _extract_string_attribute_info(attribute_series=attribute_values)
-                ]
+                attribute_info = _extract_string_attribute_info(
+                    attribute_series=attribute_values
+                )
+
+        data_dict[str(attribute)] = attribute_info
+
+    return data_dict
+
+
+def _extract_numeric_attribute_info(
+    attribute_series: pl.Series,
+) -> OrderedDict[str, MedRecordValue]:
+    """Extracts info about attributes with numeric format.
+
+    Args:
+        attribute_series (pl.Series): Series containing attribute values.
 
-        data_dict["Attribute"].extend([attribute] * len(attribute_info))
-        data_dict["Info"].extend(attribute_info)
+    Returns:
+        List[str]: attribute_info_list
+    """
+    min = attribute_series.min()
+    max = attribute_series.max()
+    mean = attribute_series.mean()
+
+    # assertion to ensure correct typing
+    assert is_medrecord_value(min)
+    assert is_medrecord_value(max)
+    assert is_medrecord_value(mean)
+
+    attribute_info = OrderedDict(
+        {
+            "min": min,
+            "max": max,
+            "mean": mean,
+        }
+    )
+
+    return attribute_info
```

Comment:
```suggestion
    return OrderedDict(
        {
            "min": min,
            "max": max,
            "mean": mean,
        }
    )
```
to satisfy above typing